src/main/java/io/avaje/config/CoreConfiguration.java

```diff
@@ -433,13 +434,14 @@ String getProperty(String key, String defaultValue) {
         // cache in concurrent map to provide higher concurrent use
         properties.put(key, val);
       }
-      return (val != NULL_PLACEHOLDER) ? val : defaultValue;
```

Comment:
The use of ` != ` instance in-equality here is intentional.  Why the change here to use `Objects.equals()`? ... can we put that back?

Reply:
I've put that back to use instance in-equality and merged into master